client/extensions/woocommerce/state/sites/coupons/actions.js

```diff
@@ -2,15 +2,18 @@
  * Internal dependencies
  */
 import {
-	WOOCOMMERCE_COUPONS_REQUEST_PAGE,
-	WOOCOMMERCE_COUPONS_PAGE_UPDATED,
+	WOOCOMMERCE_COUPONS_REQUEST,
+	WOOCOMMERCE_COUPONS_UPDATED,
 } from 'woocommerce/state/action-types';
 
-export function fetchCouponsPage( siteId, pageIndex = 1, perPage = 10 ) {
-	return { type: WOOCOMMERCE_COUPONS_REQUEST_PAGE, siteId, pageIndex, perPage };
+export function fetchCoupons( siteId, params ) {
+	// Default per_page to 10.
+	params.per_page = params.per_page || 10;
+
+	return { type: WOOCOMMERCE_COUPONS_REQUEST, siteId, params };
 }
 
-export function couponsPageUpdated( siteId, pageIndex, coupons, totalPages, totalCoupons ) {
-	return { type: WOOCOMMERCE_COUPONS_PAGE_UPDATED, siteId, pageIndex, coupons, totalPages, totalCoupons };
+export function couponsPageUpdated( siteId, params, coupons, totalPages, totalCoupons ) {
```

Comment:
Now that this is strictly tied to `page`, should the action be renamed to `couponsParamsUpdated`?

Reply:
Oh yeah, it should probably be `couponsUpdated`. Thanks, I'll change it.

Reply:
Addressed in b37e30c